src/models/graph.ts

```diff
@@ -378,6 +378,27 @@ export class Graph<N extends INodeBase, E extends IEdgeBase> extends Subject imp
   }
 
   update(data?: IObserverDataPayload): void {
+    if (data && 'type' in data && 'options' in data && 'isSingle' in data.options) {
+      if (data.type === 'node' && data.options.isSingle) {
```

Comment:
Can you maybe simplify this to `if (data && data.options?.isSingle && data.type === 'node')`?

Reply:
I believe it could be possible, but only after reworking `IObserverDataPayload` into a more generic solution as you mentioned [here](https://github.com/memgraph/orb/pull/93#discussion_r1523676765). Currently, I am using some sort of custom type guard because of the interfaces used in that payload